File: CSharpCodeBase/entities/player/player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using UnityEngine;
using System.Collections;
using MainGame.core;
//using UnityEngine.EventSystems;
 
namespace MainGame{
    public class Player : UnityEntity
    {   
        protected Vector3 homePosition;
        protected bool isStatic = false;

        public void Awake()
        {
            // FiX ME            
            //gameObject.AddComponent<PlayerController>();
            //gameObject.AddComponent<GuiControl>();
        }
        
        public void StoreHomePosition(Vector3 position)
        {
            homePosition = position;
        }         
    }
}

File: CSharpCodeBase/entities/player/player.lua
local Class = require 'utils.hump.class'
local UnityExistsEntity = require 'unity.unityentity'
local PlayerController = require 'components.character.playercontroller'
local GuiControl = require 'components.control.guicontrol'

local Player = Class({__includes = UnityExistsEntity, init = function (self, gameObject)
      UnityExistsEntity.init(self, gameObject)
      self:AddComponent(PlayerController)
      self:AddComponent(GuiControl)
      self.luaMapper:SetAlwaysVisible(true)
      self.static = false
end})

function Player:Update()
  UnityExistsEntity.Update(self)
end

function Player:FixedUpdate()
  UnityExistsEntity.FixedUpdate(self)
end

function Player:LateUpdate()
  UnityExistsEntity.LateUpdate(self)  
end

function Player:StoreHomePosition(position)
  self.homePosition = position
end

return Player